### PR TITLE
docs(java): small migration guide

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-359](DI-359)

Add a small migration guide, I'll add the breaking changes about java version in another PR.

### Changes included:

- Fix the doc to include the generic for java
- Format some tables

## 🧪 Test

open netlify

### CODE REVIEW

File: website/docs/clients/installation.mdx [hunk 1]
```diff
@@ -178,12 +178,13 @@ client.saveObject("<YOUR_INDEX_NAME>", body);
 SearchMethodParams searchMethodParams = new SearchMethodParams();
 SearchQueries requests = new SearchQueries();
 requests.setIndexName("<YOUR_INDEX_NAME>");
-requests.setQuery("my aloglia ojbect");
+requests.setQuery("my aloglia object");
```
Comment (shortcuts) [inline website/docs/clients/installation.mdx:181]: It was to highlight that we are typo tolerant ahah

**Review by shortcuts — APPROVED** [review]
Nice!

File: website/docs/clients/installation.mdx (hunk 1 above)
Comment (damcou) [reply website/docs/clients/installation.mdx:181]: He didn't even see "aloglia" OMG

**Review by damcou — APPROVED** [review]
🚀

File: website/docs/clients/installation.mdx (hunk 1 above)
Comment (millotp) [reply website/docs/clients/installation.mdx]: oops fixing it

**Review by damcou — APPROVED** [review]

**Pull request merged**